com.unity.shadergraph/Editor/Data/Nodes/AbstractMaterialNode.cs

```diff
@@ -498,17 +498,27 @@ public static ConcreteSlotValueType ConvertDynamicVectorInputTypeToConcrete(IEnu
             switch (inputTypesDistinct.Count)
             {
                 case 0:
+                    // nothing connected -- use Vec1 by default
                     return ConcreteSlotValueType.Vector1;
                 case 1:
                     if (SlotValueHelper.AreCompatible(SlotValueType.DynamicVector, inputTypesDistinct.First()))
+                    {
+                        if (inputTypesDistinct.First() == ConcreteSlotValueType.Boolean)
+                            return ConcreteSlotValueType.Vector1;
                         return inputTypesDistinct.First();
+                    }
                     break;
                 default:
                     // find the 'minumum' channel width excluding 1 as it can promote
-                    inputTypesDistinct.RemoveAll(x => x == ConcreteSlotValueType.Vector1);
+                    inputTypesDistinct.RemoveAll(x => (x == ConcreteSlotValueType.Vector1) || (x == ConcreteSlotValueType.Boolean));
                     var ordered = inputTypesDistinct.OrderByDescending(x => x);
                     if (ordered.Any())
-                        return ordered.FirstOrDefault();
+                    {
+                        var first = ordered.FirstOrDefault();
+                        if (first == ConcreteSlotValueType.Boolean)
```

Comment:
This isn't clear to me as you remove Boolean from the list but you handle the case of boolean here. Can this actually happen and if so why?

Reply:
Yeah the inner case here can be removed, as the outer case handles it.  Didn't have the outer check originally, but it's needed to make bool act like Vector1 in terms of allowing upcasts.